### PR TITLE
Grabbag of fixes for Julia 0.7 and switch to testsets

### PR DESCRIPTION
**Pull request by martinholters**
This is WIP because it cannot actually be tested on Julia 0.7 at the moment because FFTW.jl is broken there. Once that is resolved, I'll try to fix any additional problems that may have been caused in the meantime by further breaking changes or deprecations in Julia.

### CODE REVIEW

**Comment by martinholters** [comment]
With FFTW.jl fixed, I've revisited this one. Unfortunately, it now blocks on https://github.com/JuliaMath/Polynomials.jl/pull/135. But with `lastindex(::Poly)` fixed locally, tests pass on Julia 0.7 and most deprecations are also fixed. The remaining ones would be:
* `flipdim(A, d)` is deprecated, use `reverse(A, dims=d)` instead.
* `mean(A::AbstractArray, dims)` is deprecated, use `mean(A, dims=dims)` instead.
* `slicedim(A::AbstractArray, d::Integer, i)` is deprecated, use `copy(selectdim(A, d, i))` instead.

**Comment by martinholters** [comment]
I've added a workaround for the missing `lastindex(::Poly)` (by conditionally defining it) and CI is green.

I'll wait for a couple of days whether JuliaMath/Polynomials.jl#135 sees any progress and then either remove the last commit here or give it a more descriptive message. Meanwhile, this would be ready for review if anyone feels inclined to.

File: src/DSP.jl [hunk 1]
```diff
@@ -54,5 +54,12 @@ include("lpc.jl")
 using Reexport
 @reexport using .Util, .Windows, .Periodograms, .Filters, .LPC
 
+if isdefined(Base, :lastindex)
+    using Polynomials: Poly
+    if !applicable(Base.lastindex, Poly(0.0))
+        Base.lastindex(p::Poly) = endof(p)
```
Comment (ararslan) [inline src/DSP.jl]: This is type piracy

**Comment by ararslan** [comment]
https://github.com/JuliaLang/METADATA.jl/pull/13958

File: src/DSP.jl (hunk 1 above)
Comment (martinholters) [reply src/DSP.jl]: Sure. That's why the corresponding commit message says "to be deleted", which I'll now (well, on Monday, that is) do, as Polynomials has been fixed.

Comment (ararslan) [reply src/DSP.jl]: Ah sorry, didn't read the commit messages. The Polynomials tag has been merged now, btw.

**Comment by martinholters** [comment]
Removed the commit adding `lastindex`, CI is all green, all remaining deprecations warnings are due to dependencies, no further review comments to address... Finally, its time to merge this.

**Comment by ararslan** [comment]
Thanks so much!

**Comment by martinholters** [comment]
I'll tag a new release after the #189 is in.

**Pull request merged**